src/components/RepositoryNavigation.js

```diff
@@ -0,0 +1,177 @@
+import React from 'react';
+import Breadcrumb from 'react-bootstrap/Breadcrumb';
+import ListGroup from 'react-bootstrap/ListGroup';
+import ListGroupItem from 'react-bootstrap/ListGroupItem';
+import { LinkContainer } from 'react-router-bootstrap';
+import { getContents } from '../util/GithubRepositoryUtil';
+import { REPO_RESOURCE, RENDER_RESOURCE } from '../util/Constants';
+import queryString from 'query-string';
+
+/**
+ * A React component for rendering repository items when navigating the /repos resource
+ */
+class RepositoryNavigation extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      repository: {
+        isLoaded: false,
+        contents: []
+      }
+    };
+  }
+
+  /**
+   * When component first initializes, execute following lifecycle actions
+   */
+  async componentDidMount() {
+    const queryStringMap = queryString.parse(this.props.location.search);
+    const owner = queryStringMap.owner;
+    const repo = queryStringMap.repo;
+    const path = queryStringMap.path;
+
+    const json = await getContents(owner, repo, path);
+    this.setState({
+      repository: {
```

Comment:
This component is essentially the repository. This top level of the map may not be needed.

Reply:
Agreed, originally put it for semantics.